packages/esm-ward-app/src/location-selector/location-selector.component.tsx

```diff
@@ -0,0 +1,118 @@
+    const criteria = [];
+    if (debouncedSearchTerm) {
+      criteria.push(['name:contains', debouncedSearchTerm]);
+    }
+    criteria.push(['_count', size.toString()]);
+    if (emrConfiguration) {
+      criteria.push(['_tag', emrConfiguration.supportsTransferLocationTag.display]);
+    }
+    if (page > 1) {
+      criteria.push(['_getpagesoffset', ((page - 1) * size).toString()]);
+    }
+    return criteria;
+  }, [debouncedSearchTerm, page]);
+  const { locations, isLoading, totalLocations } = useLocations(filterCriteria, !emrConfiguration);
+
+  const handlePageChange = useCallback(
+    ({ page: newPage }) => {
+      setPage(newPage);
+    },
+    [setPage, page],
+  );
+  const handleSearch = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchTerm(event.target.value);
+    },
+    [setSearchTerm],
+  );
+  return (
+    <div className={styles.locationSelector}>
+      <ResponsiveWrapper>
+        <Search
+          onChange={handleSearch}
+          value={searchTerm}
+          placeholder={t('searchLocations', 'Search locations')}
+          size={isTablet ? 'lg' : 'md'}
+        />
+      </ResponsiveWrapper>
+      {isLoading || isLoadingEmrConfiguration ? (
+        <div className={styles.radioButtonGroup}>
+          <RadioButtonSkeleton />
+          <RadioButtonSkeleton />
+          <RadioButtonSkeleton />
+          <RadioButtonSkeleton />
+          <RadioButtonSkeleton />
+        </div>
+      ) : (
+        <ResponsiveWrapper>
+          <RadioButtonGroup {...props} className={styles.radioButtonGroup} value="" orientation="vertical">
+            {locations?.map((location) => (
+              <RadioButton key={location.id} labelText={location.name} value={location.id} />
+            ))}
+          </RadioButtonGroup>
+        </ResponsiveWrapper>
+      )}
+      {totalLocations > 5 && (
+        <div className={styles.pagination}>
+          <span className={styles.bodyShort01}>
+            {t('showingLocations', '{{start}}-{{end}} of {{count}} locations', {
+              start: (page - 1) * size + 1,
+              end: Math.min(page * size, totalLocations),
+              count: totalLocations,
+            })}
+          </span>
+          <div>
+            <Button
+              className={classNames(styles.button, styles.buttonLeft)}
+              disabled={page === 1}
+              onClick={() => handlePageChange({ page: page - 1 })}
+              hasIconOnly
+              renderIcon={ChevronLeftIcon}
+              kind="ghost"
+              iconDescription={t('previousPage', 'Previous page')}
+            />
+            <Button
+              className={styles.button}
+              disabled={page * size >= totalLocations}
+              onClick={() => handlePageChange({ page: page + 1 })}
+              hasIconOnly
+              renderIcon={ChevronRightIcon}
+              kind="ghost"
+              iconDescription={t('nextPage', 'Next page')}
+            />
```

Comment:
Minor, but it's generally preferable to use the [IconButton](https://react.carbondesignsystem.com/?path=/docs/components-iconbutton--overview) component for icon-only buttons https://github.com/openmrs/openmrs-esm-form-builder/pull/331